docs/internals/optimizer.rst

```diff
@@ -1382,3 +1387,55 @@ into
     }
 
 The LiteralRematerialiser should be run before this step.
+
+Codegen-Based Optimizer Module
+==============================
+
+Currently, the codegen-based optimizer module consists of only one step,
+which makes the compiler automatically generate unchecked arithmetic code that
+increments the counter variable of a `for` loop. The single step of this module,
+which is explained next, avoids wasting gas unnecessarily, by identifying some
+conditions that guarantee that the counter variable cannot overflow. This eliminates
+the need to use a explicit unchecked arithmetic block inside the loop body to
+increment the counter variable
+
+Unchecked Loop increment
+------------------------
+
+Introduced in solidity ``0.8.22``, this optimizer step is concerned with identifying
+the conditions under which the code of a ``for`` loop counter can be incremented
+without overflow checks.
+The optimization is always turned on by default and can only
+be turned off by setting the ``simpleCounterForLoopUncheckedIncrement`` flag to
+false inside ``settings.optimizer.details`` in Standard JSON input.
+It is not possible to disable it using the command-line interface.
+This optimization is only applied to `for` loops which use built-in operators ``<`` in the
+condition and (postfix/prefix) ``++`` in the lopp expression with the general form:
+
+.. code-block:: solidity
+
+    for (uint i = 0; i < x; ++i) {
+      // variable i is not modified in the loop body
+    }
+
+In many cases, the comparison is guaranteed to ensure that the counter variable
+never overflows. The precise conditions are listed and explained as follows:
+
+ - The loop condition is a comparison of the form ``i < x``, for a local counter variable ``i`` and a potentially complex expression ``x``.
+ - Built-in operator ``<`` only is used in the loop condition. User-defined operators are **not** eligible.
```

Comment:
```suggestion
 - Built-in operator ``<`` is used only in the loop condition. User-defined operators are **not** eligible.
```

Reply:
Actually, here I wanted to say that the only operator allowed in the comparison is the built-in  `<`.

Reply:
Right. The current version is good as well.